server/src/mzb_k8s_plugin.erl

```diff
@@ -67,43 +67,51 @@ create_cluster(PluginOpts, NumNodes, ClusterConfig) when is_integer(NumNodes), N
     end.
 
 destroy_cluster(ID) ->
-    R = delete_rc(ID),
-    lager:info("Destroyed RC: ~p, result: ~p", [ID, R]),
+    R = delete_deployment(ID),
+    lager:info("Destroyed Deployment: ~p, result: ~p", [ID, R]),
     {ok, _} = R,
     ok.
 
 
-%%%%%%%%%%%%%%%%%%%% 
+%%%%%%%%%%%%%%%%%%%%
 % Internal API
-%%%%%%%%%%%%%%%%%%%% 
+%%%%%%%%%%%%%%%%%%%%
 
 context_arg(undefined) -> [];
 context_arg(Context) -> ["--context", Context].
 
 namespace_arg(undefined) -> [];
 namespace_arg(Namespace) -> ["--namespace", Namespace].
 
-create_rc({Context, Namespace, BenchName}, Image, PodSpec, NumNodes) ->
+create_deployment({Context, Namespace, BenchName}, Image, PodSpec, NumNodes) ->
+    JsonFile = write_json_resource(BenchName),
+    {ok, Binary} = file:read_file(JsonFile),
+
     % Might be a good idea to be able to define these values in the benchmark
     CPU = get_config_value(cpu, PodSpec),
     Memory = get_config_value(memory, PodSpec),
 
     Resources = "cpu=" ++ CPU ++ ",memory=" ++ Memory,
-    Cmd = ["kubectl", "run", BenchName] ++
+
+    Cmd = ["kubectl", "run", BenchName ++ "-worker"] ++
           context_arg(Context) ++
           namespace_arg(Namespace) ++
           ["--labels", "app=mzbench-worker,bench=" ++ BenchName,
            "--image", Image,
            "--replicas", integer_to_list(NumNodes),
            "--requests", Resources,
-           "--limits", Resources,
            "--port", "22",
-           "--generator", "run/v1",
+           "--overrides", "'" ++ binary_to_list(Binary) ++ "'",
+           "--image-pull-policy", "Always",
            "--", "/usr/sbin/sshd", "-D"],
-    run_kubectl(Cmd, false).
-
-delete_rc({Context, Namespace, BenchName}) ->
-    Cmd = ["kubectl", "delete", "rc", BenchName] ++
+    X = run_kubectl(Cmd, false),
+    io:format("~p~n", [X]),
+    {ok, _} = X,
```

Comment:
1. We print everything with the help of `lager`.
2. The `run_kubectl` already do log the command itself so we might consider printing it there.
3. It might be a temporal debug print at all?